client/main.lua

```diff
@@ -366,9 +378,15 @@ RegisterNUICallback('select', function(data, cb)
         elseif option.command then
             ExecuteCommand(option.command)
         end
+        
+        if option.openMenu ~= currentMenu then
+            currentMenu = option.openMenu
+            state.setNuiFocus(true, true, false)
+            forceNuiActive = true
+        end
     end
 
-    if IsNuiFocused() then
+    if IsNuiFocused() and not forceNuiActive then
         state.setActive(false)
     end
 end)
```

Comment:
I'm a bit apprehensive of triggering any actions when also opening a "menu" but could see situations where it's needed. The following should work a lot better than your "forceNui" approach.

Can also set currentMenu as `nil` here rather than needing to set `'home'` on everything.

```lua
RegisterNUICallback('select', function(data, cb)
    cb(1)

    local option = options?[data[1]][data[2]]

    if option then
        if option.openMenu then
            currentMenu = option.openMenu ~= 'home' and option.openMenu or nil
        else
            state.setNuiFocus(false)
        end

        if option.onSelect then
            option.onSelect(option.qtarget and currentTarget.entity or getResponse(option))
        elseif option.export then
            exports[option.resource][option.export](nil, getResponse(option))
        elseif option.event then
            TriggerEvent(option.event, getResponse(option))
        elseif option.serverEvent then
            TriggerServerEvent(option.serverEvent, getResponse(option, true))
        elseif option.command then
            ExecuteCommand(option.command)
        end
    end

    if not option.openMenu and IsNuiFocused() then
        state.setActive(false)
    end
end)
```